index.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const { inherits, format } = require('util')
+const codes = {}
```

Comment:
I'm heavily -1 for this to be a singleton. This depends on the module resolution algorithm to always resolve to the same instance of this module, and we cannot assure that. I'm -1.

`createWarning` should not store this there.

An alternative API would be:

```js
const warnings = new Warnings()
warnings.createWarning()
warnings.emitWarning()
```

In this way the consumer is responsible for keeping a storage of those.

```

Reply:
Agreed.

Reply:
FYI: I have made the mistake of creating singleton exporting modules in the past. It's not fun.